qiita_pet/handlers/study_handlers/sample_template.py

```diff
@@ -57,3 +61,50 @@ def post(self):
         else:
             raise HTTPError(400, 'Unknown sample template action: %s' % action)
         self.write(result)
+
+
+def build_sample_summary(study_id, user_id, visible=None):
+    if visible is None:
+        visible = []
+    # Load sample template into dataframe and filter to wanted columns
+    df = pd.DataFrame.from_dict(
+        sample_template_get_req(int(study_id), user_id)['template'],
+        orient='index', dtype=str)
+    meta_available = set(df.columns)
+    # Add one column per prep template highlighting what samples exist
+    prep_cols = []
+    preps = study_prep_get_req(study_id, user_id)['info']
+    for dt in preps:
+        for prep in preps[dt]:
+            prep_samples = prep_template_get_req(
+                prep['id'], user_id)['template'].keys()
+            prep_df = pd.Series(['X'] * len(prep_samples),
+                                index=prep_samples, dtype=str)
+            col_name = ' - '.join([prep['name'], str(prep['id'])])
+            prep_cols.append(col_name)
+            df[col_name] = prep_df
+
+    # Format the dataframe to html table
+    meta_available = meta_available.difference(visible)
+    table = df.to_html(classes='table table-striped', na_rep='',
+                       columns=prep_cols + visible)
+    return table, meta_available
+
+
+class SampleAJAX(BaseHandler):
+    def get(self):
```

Comment:
I'm having problems understanding what the parameters of this Handler means/do. Can you add more comments to the code?